internal/deviceplugin/server.go

```diff
@@ -125,7 +127,10 @@ func (srv *server) Allocate(ctx context.Context, rqt *pluginapi.AllocateRequest)
 				if cresp.Envs == nil {
 					cresp.Envs = make(map[string]string)
 				}
-				cresp.Envs[key] = value
+				re := regexp.MustCompile(`[0-9]+`)
+				str := strconv.Itoa(len(cresp.Envs))
+				newkey := re.ReplaceAllString(key, str)
+				cresp.Envs[newkey] = value
```

Comment:
How about a variable called `FPGA_DCP11_DEV0` if such was set in the FPGA plugin? This code change would definitely break the plugin. I believe this logic is specific to QAT. For this purpose the QAT's `devicePlugin` can implement the optional `PostAllocator` interface to rename the variables. Look at `devicePlugin.PostAllocate()` in the FPGA plugin for an example.

Reply:
We can always change the regular expression to check for numbers only at the end of the environment variable name? That would comply with FPGA too right? I think its important to have suitable environment variables for all the plugins eg if 3 QAT devices and 3 FPGA devices are requested by a container, environment variable should be predictable eg QAT0, QAT1 and QAT2 or FPGA_DCP11_DEV0 FPGA_DCP11_DEV1 and FPGA_DCP11_DEV2. I am not sure what the environment variables names are in case of FPGA currently but in case of QAT the variables are random based on the key of the map which is used to store the device information which means that the environment variable could be something like QAT21, QAT5, QAT25 which is not very user friendly hence the proposed PR. Hopefully, I am able to explain the motivation behind this PR clearly and the fact that this is applicable to all the plugins and not just QAT.

Reply:
Thanks for directing me to the postallocator interface, I will have a look and see if that's a better way of implementing this.

Reply:
I'm sure I understand your motivation and I had in mind exactly this concern about user friendliness too. Just wanted to emphasize that this change
1. makes the assumption it won't break other plugins;
2. introduces new constrains on how environment variables used by other plugins can be named.

For example even with the corrected regexp variables like `SOME_CFG_PARAM_FOR_VERSION_7` become illegitimate and all the developers have to be aware of this constraint. To adhere to the principle of least astonishment I'd rather let them assume the variables they set are not going to be renamed by some underlying layer.

Reply:
Cool, thanks for explaining and providing details on this. Probably the postallocator mentioned by you is the best option then, I will reimplement this PR using that.